src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationParsingTests.cs

```diff
@@ -6333,5 +6333,129 @@ class C<T> : where T : X
             }
             EOF();
         }
+
+        [Fact]
+        [WorkItem(23833, "https://github.com/dotnet/roslyn/issues/23833")]
+        public void ProduceErrorsOnRef_Properties_Ref_Get()
+        {
+            var code = @"
+class Program
+{
+    public int P
+    {
+        ref get => throw null;
+    }
+}";
+
+            CreateStandardCompilation(code).VerifyDiagnostics(
+                // (5,6): error CS1513: } expected
```

Comment:
>// (5,6): error CS1513: } expected [](start = 16, length = 34)

Perhaps the experience would be better if we parse ```ref``` as a modifier, but report an "invalid modifier" error for it and ignore it during semantic analysis.

Reply:
I also wonder if error recovery would be better if we just accept `ref` in parser for this case and give an error in binding.

---
In reply to: [158535495](https://github.com/dotnet/roslyn/pull/23905#discussion_r158535495) [](ancestors = 158535495)

Reply:
I tried that yesterday. The main problem with this approach is that if `ref` is parsed freely as a modifier (without specifically checking if there is a `struct` keyword after it), it will wrongly parse `ref type identifier` wrongly as a by-val variable/field declaration, with `ref` being a modifier of that declaration.
The parser can be modified to accept `ref` in certain situations that we know for sure, but I believe this is not a common enough position to do so. Please let me know if you see otherwise.

---
In reply to: [158571025](https://github.com/dotnet/roslyn/pull/23905#discussion_r158571025) [](ancestors = 158571025,158535495)

Reply:
I meant that we would still parse accessors as accessors even when a ```ref``` modifier is present. We would still report an error, probably a new, very specific error saying that a ```ref``` modifier or keyword cannot be used here.

---
In reply to: [158574706](https://github.com/dotnet/roslyn/pull/23905#discussion_r158574706) [](ancestors = 158574706,158571025,158535495)